File: app/src/main/java/com/bevstudio/wolfbooksapp/request/api/RequestService.java
package com.bevstudio.wolfbooksapp.request.api;

import com.bevstudio.wolfbooksapp.model.api.Books;
import com.bevstudio.wolfbooksapp.model.api.Item;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface RequestService {
    @GET("/books/v1/volumes")
    Call<Books> getVolumeBooks(@Query(value = "q") String searchText);

    @GET("/books/v1/volumes/{id}")
    Call<Item> getBookItem(@Path("id") String id);

    @GET("/books/v1/volumes?q=categories:young+fiction&maxResults=30")
    Call<Books> getNewReleaseBooks();

    @GET("/books/v1/volumes")
    Call<Books> getCategories(@Query(value = "q") String searchText, @Query("filter") String filter, @Query("orderBy") String orderBy, @Query("maxResults") int maxResults);


    @GET("/books/v1/volumes")
    Call<Books> getSearchResults(@Query(value = "q") String searchText, @Query("startIndex") int startIndex, @Query("orderBy") String orderBy, @Query("maxResults") int maxResults);
}

File: app/src/main/java/com/bevstudio/wolfbooksapp/request/api/RequestService.kt
package com.bevstudio.wolfbooksapp.request.api

import com.bevstudio.wolfbooksapp.model.api.Books
import com.bevstudio.wolfbooksapp.model.api.Item
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface RequestService {
    @GET("/books/v1/volumes")
    fun getVolumeBooks(@Query(value = "q") searchText: String?): Call<Books>

    @GET("/books/v1/volumes/{id}")
    fun getBookItem(@Path("id") id: String?): Call<Item>

    @get:GET("/books/v1/volumes?q=categories:young+fiction&maxResults=30")
    val newReleaseBooks: Call<Books>

    @GET("/books/v1/volumes")
    fun getCategories(
        @Query(value = "q") searchText: String?,
        @Query("filter") filter: String?,
        @Query("orderBy") orderBy: String?,
        @Query("maxResults") maxResults: Int
    ): Call<Books>


    @GET("/books/v1/volumes")
    fun getSearchResults(
        @Query(value = "q") searchText: String?,
        @Query("startIndex") startIndex: Int,
        @Query("orderBy") orderBy: String?,
        @Query("maxResults") maxResults: Int
    ): Call<Books>
}
